internal/core/service/interfaces.go

```diff
@@ -0,0 +1,49 @@
+package service
+
+//go:generate mockgen -source=$GOFILE -destination=mocks/services.go
+
+import (
+	"context"
+	"github.com/zhuravlev-pe/course-watch/internal/core/domain"
+	"github.com/zhuravlev-pe/course-watch/internal/core/dto"
+)
+
+type UsersRepository interface {
+	GetById(ctx context.Context, id string) (*domain.User, error)
+	Insert(ctx context.Context, user *domain.User) error
+	Update(ctx context.Context, id string, input *dto.UpdateUserInfoInput) error
+	GetByEmail(ctx context.Context, email string) (*domain.User, error)
+}
+
+type CoursesRepository interface {
+	GetById(ctx context.Context, id string) (*domain.Course, error)
+	Insert(ctx context.Context, course *domain.Course) error
+}
+
+type Users interface {
+	GetUserInfo(ctx context.Context, id string) (*dto.GetUserInfoOutput, error)
+	UpdateUserInfo(ctx context.Context, id string, input *dto.UpdateUserInfoInput) error
+	Login(ctx context.Context, input *dto.LoginInput) (*domain.User, error)
+	Signup(ctx context.Context, input *dto.SignupUserInput) error
+}
+
+type Courses interface {
+	GetById(ctx context.Context, id string) (*domain.Course, error)
+	Create(ctx context.Context, input dto.CreateCourseInput) (*domain.Course, error)
+}
+
```

Comment:
Apparently, community has no strict rule for that. For me, personally, `ID` is more eye pleasing, but, since everywhere in the project is used `Id`, we should be consistent. 

In addition I've learned that neither "TCP" nor "id" are acronyms and both are abbreviations - a shortened form of a word or phrase :)

Reply:
This is an extensive discussion (https://github.com/golang/lint/issues/124), bigger than it should be for my taste. What is important is to be consistent with the decision that we make